lib/maths/CDataFrameUtils.cc

```diff
@@ -74,5 +75,50 @@ bool CDataFrameUtils::standardizeColumns(std::size_t numberThreads, core::CDataF
 
     return frame.writeColumns(numberThreads, standardiseColumns);
 }
+
+bool CDataFrameUtils::columnQuantiles(std::size_t numberThreads,
+                                      const core::CDataFrame& frame,
+                                      const TSizeVec& columnMask,
+                                      TQuantileSketchVec& result,
+                                      TWeightFunction weight) {
+    if (result.size() != columnMask.size()) {
+        return false;
```

Comment:
It's probably a good idea to log an error with the 2 different sizes here (a) to give more information to debug it if it does ever happen in production (I assume it should be impossible if the code works as intended) and (b) to distinguish from the other code path that returns `false` (which I assume is more likely).